orchestrator/src/itn_orchestrator/main.go

```diff
@@ -269,15 +272,26 @@ func main() {
 			continue
 		}
 		cmd := *commandOrComment.command
-		if prevAction != nil && prevAction.Name() != cmd.Action {
-			handlePrevAction()
-		}
 		params, err := lib.ResolveParams(rconfig, step, cmd.Params)
 		if err != nil {
 			log.Errorf("Error resolving params for step %d: %v", step, err)
 			os.Exit(6)
 			return
 		}
+		if cmd.Action == "fund-keys" {
+			var fundParams itn_orchestrator.FundParams
```

Comment:
I suggest that we move this code back to `fund.go` in the following way:

1. Add a `Validate` method to `BatchAction` interface
2. Implement `Validate` for `FundAction` by copying the code in the current `if cmd.Action == "fund-keys" { ... }` section
3. Run `Validate` before line `prevAction = batchAction`

This way we'll ensure both early exit and that the code in `main.go` remains fully generic

Reply:
Done as part of the https://github.com/o1-labs/mina-perf-testing/pull/2/commits/34d17cc0733837c3d557bcf7963ff426064246d1